CMakeLists.txt

```diff
@@ -1,506 +1,2043 @@
+   if needed
+#]===================================================================]
+
+if (static)
+  set (Protobuf_USE_STATIC_LIBS ON)
+endif ()
+find_package (Protobuf)
+if (MSVC AND NOT TARGET protobuf::protoc)
+  find_program (my_protoc NAMES protoc)
+  if (NOT my_protoc)
+    message (WARNING "No protoc found -- downloading now")
+    file (DOWNLOAD
+      "https://ripple.github.io/Downloads/protoc/2.5.1/protoc.exe"
+      ${CMAKE_BINARY_DIR}/tools/protoc.exe
+      STATUS status
+      TIMEOUT 30)
+    list (GET status 0 status_code)
+    list (GET status 1 status_string)
+    if (NOT status_code EQUAL 0)
+      message (FATAL_ERROR "Failed to download protoc: ${status_string}")
+    endif ()
+    set (my_protoc ${CMAKE_BINARY_DIR}/tools/protoc.exe)
+  endif ()
+  add_executable (protobuf::protoc IMPORTED)
+  set_target_properties (protobuf::protoc PROPERTIES
+    IMPORTED_LOCATION "${my_protoc}")
+endif ()
+
+file (MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/proto_gen)
+set (save_CBD ${CMAKE_CURRENT_BINARY_DIR})
+set (CMAKE_CURRENT_BINARY_DIR ${CMAKE_BINARY_DIR}/proto_gen)
+protobuf_generate_cpp (
+  PROTO_SRCS
+  PROTO_HDRS
+  src/ripple/proto/ripple.proto)
+set (CMAKE_CURRENT_BINARY_DIR ${save_CBD})
+
+add_library (pbufs STATIC ${PROTO_SRCS} ${PROTO_HDRS})
+if ((NOT TARGET protobuf::libprotobuf) AND (NOT Protobuf_LIBRARIES))
+  message (STATUS "protobuf lib not found - adding unity source for lib")
+  target_sources (pbufs PRIVATE src/ripple/unity/protobuf.cpp)
+  target_include_directories (pbufs
+    PUBLIC
+      src/protobuf/src
+      src/protobuf/vsprojects
+      src/ripple/proto)
+  target_compile_definitions (pbufs
+    PRIVATE
+      _SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS)
+endif ()
+target_include_directories (pbufs PRIVATE src)
+target_include_directories (pbufs
+  SYSTEM PUBLIC
+    ${CMAKE_BINARY_DIR}/proto_gen)
+if (NOT MSVC)
+  if (TARGET protobuf::libprotobuf)
+    target_link_libraries (pbufs protobuf::libprotobuf)
+  else ()
+    target_link_libraries (pbufs ${Protobuf_LIBRARIES})
+    target_include_directories (pbufs PUBLIC ${Protobuf_INCLUDE_DIR})
+  endif ()
+endif ()
+target_compile_options (pbufs
+  PUBLIC
+    $<$<BOOL:${is_xcode}>:
+      --system-header-prefix="google/protobuf"
+      -Wno-deprecated-dynamic-exception-spec
+    >
+    $<$<BOOL:${MSVC}>:
+      -wd4251
+      -wd4146
+    >
+  PRIVATE
+    $<$<BOOL:${MSVC}>:
+      -wd4018
+      -wd4244
+      -wd4800
+      -wd4996
+    >)
+add_library (Ripple::pbufs ALIAS pbufs)
+target_link_libraries (ripple_libs INTERFACE Ripple::pbufs)
+
+#[===================================================================[
+   create a lib to separate/isolate the legacy beast code
+   TODO: continue to review these sources for removal or replacement
+#]===================================================================]
+
+file (GLOB_RECURSE rb_headers
+  src/ripple/beast/*.h
+  src/ripple/beast/*.hpp)
+
+add_library (xrpl_legacy
```

Comment:
Names are hard. I don't love `xrpl_legacy`. No change needed, just registering that we could improve on the `legacy` name if someone had a better idea. Maybe `xrpl_beast`?

Reply:
Since you are probably the only ones using this library and maintaining it, why not completely rename it to something like `libvinnie`?

Reply:
yeah, naming is hard ;). Another option is to just fold these files into `xrpl_core` since they are really just the legacy parts of core. The physical separation into two libraries was just to make it easier to replace/remove/deprecate parts of `beast/legacy/vinnie` moving forward, but I'm not sure it's worth the extra component at this point